components/supervisor/cmd/init.go

```diff
@@ -83,7 +83,11 @@ var initCmd = &cobra.Command{
 			if err != nil && !(strings.Contains(err.Error(), "signal: ") || strings.Contains(err.Error(), "no child processes")) {
 				if eerr, ok := err.(*exec.ExitError); ok && eerr.ExitCode() != 0 {
 					logs := extractFailureFromRun()
-					log.WithError(fmt.Errorf(logs)).Fatal("supervisor run error with unexpected exit code")
+					if eerr.ExitCode() == 2 {
```

Comment:
I'm not sure why we need this change?

Reply:
it looks better than generic error :+1 

where do we set exitCode to 2?

Reply:
Can we get rid of `supervisor run error`?

Reply:
There's a line on supervisor.go `shutdown <- ShutdownReasonIDEReadinessTimedOut`

We could make it shared as well like that metric timeout one

Reply:
> Can we get rid of `supervisor run error`? and make error after `:` more specific, like `IntelliJ backend timed out to start after 5 minutes`?

Looks more reasonable 👍